orangecontrib/educational/widgets/owpolynomialregression.py

```diff
@@ -177,22 +178,27 @@ def clear_plot(self):
     @check_sql_input
     def set_data(self, data):
         self.clear()
-        self.data = data
+        self.Error.no_cont_variables.clear()
         if data is not None:
             cvars = [var for var in data.domain.variables if var.is_continuous]
             class_cvars = [var for var in data.domain.class_vars
                            if var.is_continuous]
 
-            self.x_var_model[:] = cvars
-            self.y_var_model[:] = cvars
-
             nvars = len(cvars)
             nclass = len(class_cvars)
+            if nvars == 0:
```

Comment:
When no numeric features `self.x_var_model[:]` and `self.y_var_model[:]§` should be set to `[]`. 
Moving entire `if` clause two lines lower would solve a problem.